### PR TITLE
Issu831 weather agent

### LINKED ISSUE #831: WeatherAgent does not work properly with country/city instead of zip_code
The code of weatherAgent is not correct to use it with country/city instead of zip_code.
The following lines of the file: https://github.com/VOLTTRON/volttron/blob/master/services/core/WeatherAgent/weather/weatheragent.py

```
    agent_id = get_config('agentid')
    poll_time = get_config('poll_time')
    zip_code = get_config("zip")
    key = get_config('key')
    on_request_only = get_config('on_request_only')

    state = ''
    country = ''
    city = ''
    region = state if state != "" else country
    city = city
```

should be replaced by:

```
agent_id = get_config('agentid')
    poll_time = get_config('poll_time')
    zip_code = get_config("zip")
    state = get_config("state")
    country = get_config("country")
    city = get_config("city")
    key = get_config('key')
    on_request_only = get_config('on_request_only')
```

and also:

```
elif self.region != ""
```

by

```
elif region !=""
```

### PR DESCRIPTION
**Pull request by damienpicard**
This fixes #831

---
This change is [Reviewable](https://reviewable.io/reviews/volttron/volttron/1047)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -19,3 +19,4 @@ parser.out
 /config/
 /docs/build/*
 /docs/source/apidocs
+/applications/*
```
Comment (craig8) [inline .gitignore]: I shoul have done this  !

File: services/core/WeatherAgent/weather/weatheragent.py [hunk 2]
```diff
@@ -174,25 +174,24 @@ def weather_service(config_path, **kwargs):
     config = utils.load_config(config_path)
 
     def get_config(name):
-        try:
-            return kwargs.pop(name)
-        except KeyError:
-            return config.get(name, '')
+		try:
```
Comment (craig8) [inline services/core/WeatherAgent/weather/weatheragent.py]: Why are there so many spaces here?

Comment (damienpicard) [inline services/core/WeatherAgent/weather/weatheragent.py]: Sorry for that, we use a different indentation incoding!

**Comment by kmonson** [comment]
Superseded by https://github.com/VOLTTRON/volttron/pull/1414

**Pull request closed without merging**